src/pool/srv_pool.c

```diff
@@ -5918,6 +6048,36 @@ pool_svc_update_map_internal(struct pool_svc *svc, unsigned int opc,
 	if (map_version == map_version_before)
 		D_GOTO(out_map, rc = 0);
 
+	/*
+	 * If the map modification affects myself, leave it to a new PS leader
+	 * if there's another PS replica, or reject it.
+	 */
+	rc = pool_map_find_nodes(map, dss_self_rank(), &node);
```

Comment:
not very sure if using "dss_self_rank()" is always correct, for example if the pool is created over ranks 1,3,5,7 out of total 8 ranks (0 - 7), the 4 ranks in the pool's sub-group will with logical rank 0 - 3?
But if with problem there are some other places also with the issue.

Reply:
Thanks, Xuezhao. Do you mean I should have used `pool_map_find_node_by_rank` instead? I think you're right.

Ranks in pool map are PG ranks, not SG ranks. If I'm not mistaken, I think using `dss_self_rank` is correct.

Reply:
OK, using dss_self_rank is correct, thx. And your change to pool_map_find_node_by_rank() is also correct although it is not my original intended suggestion, but happened to find an issue, :)